client-spark/spark3/src/main/java/org/apache/spark/shuffle/HybridShuffleManager.java

```diff
@@ -0,0 +1,386 @@
+                    (ShuffleReader<K, C>)
+                            rssShuffleManager
+                                    .getClass()
+                                    .getDeclaredMethod(
+                                            "getReaderForRange",
+                                            ShuffleHandle.class,
+                                            int.class,
+                                            int.class,
+                                            int.class,
+                                            int.class,
+                                            TaskContext.class,
+                                            ShuffleReadMetricsReporter.class)
+                                    .invoke(
+                                            handle,
+                                            startMapIndex,
+                                            endMapIndex,
+                                            startPartition,
+                                            endPartition,
+                                            context,
+                                            metrics):
+                    (ShuffleReader<K, C>)
+                            essShuffleManager
+                                    .getClass()
+                                    .getDeclaredMethod(
+                                            "getReaderForRange",
+                                            ShuffleHandle.class,
+                                            int.class,
+                                            int.class,
+                                            int.class,
+                                            int.class,
+                                            TaskContext.class,
+                                            ShuffleReadMetricsReporter.class)
+                                    .invoke(
+                                            handle,
+                                            startMapIndex,
+                                            endMapIndex,
+                                            startPartition,
+                                            endPartition,
+                                            context,
+                                            metrics);
+        } catch (Exception e) {
+            throw new RssException(e);
+        }
+        return reader;
+    }
+
+    @Override
+    public boolean unregisterShuffle(int shuffleId) {
+        return rssShuffleIds.contains(shuffleId) ?
+                rssShuffleManager.unregisterShuffle(shuffleId):
+                essShuffleManager.unregisterShuffle(shuffleId);
+    }
+
+    @Override
+    public void stop() {
+        essShuffleManager.stop();
+        if (rssShuffleManager != null) {
+            rssShuffleManager.stop();
+        }
+        coordinatorClients.forEach(CoordinatorClient::close);
+    }
+
+    @Override
+    public ShuffleBlockResolver shuffleBlockResolver() {
```

Comment:
Hi, @zuston ,

How can we resolve [shuffleBlockResolver](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/shuffle/ShuffleBlockResolver.scala)?
It seems shuffleId can not be accessed?

Reply:
RSS needn't this. So if the ess is initialized in this app lifecycle, please delegate it.

Reply:
```
  @Override
  public ShuffleBlockResolver shuffleBlockResolver() {
    return getOrCreateESS().shuffleBlockResolver();
  }
```